packages/plugin-workspace-tools/sources/commands/foreach.ts

```diff
@@ -126,14 +134,22 @@ export default class WorkspacesForeachCommand extends BaseCommand {
       ? project.topLevelWorkspace
       : cwdWorkspace!;
 
+    const rootCandidates = this.since
+      ? Array.from(await gitUtils.fetchChangedWorkspaces({ref: this.since, project}))
+      : [rootWorkspace, ...(this.from.length > 0 ? rootWorkspace.getRecursiveWorkspaceChildren() : [])];
```

Comment:
Note to self: might be worth later to find a good semantic for `--from --since`

Reply:
I think throwing is probably the right thing to do, at least for now. Quoting myself from https://github.com/yarnpkg/berry/issues/3450:

> The interaction of `--since` with `--from`/`--all` is not clear, and I would be tempted to consider providing both an illegal invocation.